db/migrations/00012_add_pending_diff_status.sql

```diff
@@ -0,0 +1,53 @@
+-- +goose Up
+DROP INDEX storage_diff_new_status_index;
+DROP INDEX storage_diff_unrecognized_status_index;
+
+ALTER TABLE public.storage_diff ALTER COLUMN status DROP DEFAULT;
+ALTER TABLE public.storage_diff ALTER COLUMN status TYPE VARCHAR(255);
+
+DROP TYPE public.diff_status;
+CREATE TYPE public.diff_status AS ENUM (
+    'new',
+    'pending',
+    'transformed',
+    'unrecognized',
+    'noncanonical',
+    'unwatched'
+    );
+
+ALTER TABLE public.storage_diff ALTER COLUMN status TYPE public.diff_status USING (status::diff_status);
+ALTER TABLE public.storage_diff ALTER COLUMN status SET DEFAULT 'new';
+
+CREATE INDEX storage_diff_new_status_index
+    ON public.storage_diff (status) WHERE status = 'new';
+CREATE INDEX storage_diff_unrecognized_status_index
+    ON public.storage_diff (status) WHERE status = 'unrecognized';
+
+CREATE INDEX storage_diff_pending_status_index
+    ON public.storage_diff (status) WHERE status = 'pending';
+
+-- +goose Down
+UPDATE public.storage_diff SET status = 'new' WHERE status = 'pending';
+DROP INDEX storage_diff_new_status_index;
+DROP INDEX storage_diff_unrecognized_status_index;
```

Comment:
The problem is the indexes point to (per the schema)

`CREATE INDEX storage_diff_new_status_index ON public.storage_diff USING btree (status) WHERE (status = 'new'::public.diff_status);`

So when you try and do the `ALTER TABLE public.storage_diff ALTER COLUMN status TYPE VARCHAR(255);` it fails, because the index is looking for a public.diff_status type not a varchar. I haven't been able to find a way to alter the indexes where clause in place, unfortunately.

Reply:
Makes sense, thanks for trying anyway